leafmap/common.py

```diff
@@ -10917,3 +10917,268 @@ def index():
     else:
         # Run the Flask server in the main thread
         run_flask()
+
+
+def vector_to_mbtiles(
+    input_file: str,
+    output_file: str,
+    layer_name: Optional[str] = None,
+    options: Optional[List[str]] = None,
+    quiet: bool = False,
+) -> Optional[str]:
+    """
+    Converts vector data to .mbtiles using Tippecanoe.
+
+    Args:
+        input_file (str): Path to the input vector data file (e.g., .geojson).
+        output_file (str): Path to the output .mbtiles file.
+        layer_name (Optional[str]): Optional name for the layer. Defaults to None.
+        options (Optional[List[str]]): List of additional arguments for tippecanoe. Defaults to None.
+        quiet (bool): If True, suppress the log output. Defaults to False.
+
+    Returns:
+        Optional[str]: Output from the Tippecanoe command, or None if there was an error or if Tippecanoe is not installed.
+
+    Raises:
+        subprocess.CalledProcessError: If there's an error executing the tippecanoe command.
+    """
+
+    import subprocess
+    import shutil
+
+    # Check if tippecanoe exists
+    if shutil.which("tippecanoe") is None:
+        print("Error: tippecanoe is not installed.")
+        print("You can install it using conda with the following command:")
+        print("conda install -c conda-forge tippecanoe")
+        return None
+
+    command = ["tippecanoe", "-o", output_file]
+
+    # Add layer name specification if provided
+    if layer_name:
+        command.extend(["-L", f"{layer_name}:{input_file}"])
+    else:
+        command.append(input_file)
+
+    # Append additional arguments if provided
+    if options:
+        command.extend(options)
+
+    try:
+        process = subprocess.Popen(
+            command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True
+        )
+
+        if not quiet:
+            for line in process.stdout:
+                print(line, end="")
+
+        exit_code = process.wait()
+        if exit_code != 0:
+            raise subprocess.CalledProcessError(exit_code, command)
+
+    except subprocess.CalledProcessError as e:
+        print(f"\nError executing tippecanoe: {e}")
+        return None
+
+    return "Tippecanoe process completed successfully."
+
+
+def mbtiles_to_pmtiles(
```

Comment:
I do not recommend using the Python package to convert mbtiles -> pmtiles. You should check that the tippecanoe version is >2.0 and create pmtiles directly.

Reply:
I thought there were issues with tippecanoe when converting mbtiles to pmtiles based on the comments [here](https://github.com/tebben/overture-maps/blob/main/scripts/convert/overture/geojson_to_pmtiles.sh#L6). I just tested it again and it seems now tippecanoe can convert mbtiles to pmtiles v3 successfully. I will switch to tippecanoe to create pmtiles directly. Thanks.